### PR TITLE
Fix download leak

### PR DESCRIPTION
**Pull request by doncung**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @doncung has signed the CLA. Thanks for the pull request!

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxHttpResponse.java [hunk 1]
```diff
@@ -198,7 +202,7 @@ private String readStream(InputStream inputStream) throws IOException, BoxExcept
                 builder.append(buffer, 0, read);
                 read = reader.read(buffer, 0, BUFFER_SIZE);
             }
-
+            disconnect();
```
Comment (seema-at-box) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxHttpResponse.java]: finally

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestDownload.java [hunk 2]
```diff
@@ -354,15 +369,30 @@ public File getOutputFile() {
                         output = getOutputStream(downloadInfo);
                     }
                     SdkUtils.copyStream(response.getBody(), output);
+
                 } catch (Exception e) {
+                    // For zip encoded downloads we must kill the socket or it will leak.
+                    Socket socket = mRequest.getSocket();
+                    if (socket != null && contentEncoding != null && contentEncoding.equalsIgnoreCase("gzip")) {
```
Comment (seema-at-box) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestDownload.java]: constant

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxHttpResponse.java (hunk 1 above)
Comment (doncung) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxHttpResponse.java]: This cannot be called after reader.close the code is essentially inherited from:
https://github.com/box/box-java-sdk/blob/94ce2b691b7faa47835367bc55b371e62b779005/src/main/java/com/box/sdk/BoxJSONResponse.java

**Comment by seema-at-box** [comment]
+1

**Pull request merged**